modules/tensor_mechanics/src/utils/GuaranteeProvider.C

```diff
@@ -27,3 +27,11 @@ GuaranteeProvider::issueGuarantee(const MaterialPropertyName & prop_name, Guaran
   // intentional insertion
   _guarantees[prop_name].insert(guarantee);
 }
+
+void
+GuaranteeProvider::revokeGuarantee(const MaterialPropertyName & prop_name, Guarantee guarantee)
+{
+  auto it = _guarantees.find(prop_name);
+  if (it != _guarantees.end())
+    it->second.erase(guarantee);
```

Comment:
@dschwen If this is the last `guarantee` for a particular `prop_name` do you also want to remove `prop_name` from the map entirely, or is it OK for it to be there with an empty `std::set`?